Gateway fails to start in tests with address already in use

```
java.lang.RuntimeException: Gateway was not able to start
	at io.zeebe.broker.clustering.base.EmbeddedGatewayService.startGateway(EmbeddedGatewayService.java:64) ~[classes/:?]
	at io.zeebe.util.sched.RunnableAdapter.lambda$wrapRunnable$0(RunnableAdapter.java:48) ~[zeebe-util-0.18.0-SNAPSHOT.jar:0.18.0-SNAPSHOT]
	at io.zeebe.util.sched.RunnableAdapter.run(RunnableAdapter.java:35) ~[zeebe-util-0.18.0-SNAPSHOT.jar:0.18.0-SNAPSHOT]
	at io.zeebe.util.sched.BlockingPollSubscription.run(BlockingPollSubscription.java:55) ~[zeebe-util-0.18.0-SNAPSHOT.jar:0.18.0-SNAPSHOT]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_212]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_212]
	at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_212]
Caused by: java.io.IOException: Failed to bind
	at io.grpc.netty.NettyServer.start(NettyServer.java:251) ~[grpc-netty-1.20.0.jar:1.20.0]
	at io.grpc.internal.ServerImpl.start(ServerImpl.java:177) ~[grpc-core-1.20.0.jar:1.20.0]
	at io.grpc.internal.ServerImpl.start(ServerImpl.java:85) ~[grpc-core-1.20.0.jar:1.20.0]
	at io.zeebe.gateway.Gateway.start(Gateway.java:89) ~[zeebe-gateway-0.18.0-SNAPSHOT.jar:0.18.0-SNAPSHOT]
	at io.zeebe.broker.clustering.base.EmbeddedGatewayService.startGateway(EmbeddedGatewayService.java:62) ~[classes/:?]
	... 6 more
Caused by: java.net.BindException: Address already in use
	at sun.nio.ch.Net.bind0(Native Method) ~[?:1.8.0_212]
	at sun.nio.ch.Net.bind(Net.java:433) ~[?:1.8.0_212]
	at sun.nio.ch.Net.bind(Net.java:425) ~[?:1.8.0_212]
	at sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:223) ~[?:1.8.0_212]
	at io.netty.channel.socket.nio.NioServerSocketChannel.doBind(NioServerSocketChannel.java:132) ~[netty-transport-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.bind(AbstractChannel.java:563) ~[netty-transport-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.bind(DefaultChannelPipeline.java:1332) ~[netty-transport-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeBind(AbstractChannelHandlerContext.java:503) ~[netty-transport-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.channel.AbstractChannelHandlerContext.bind(AbstractChannelHandlerContext.java:488) ~[netty-transport-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.channel.DefaultChannelPipeline.bind(DefaultChannelPipeline.java:984) ~[netty-transport-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.channel.AbstractChannel.bind(AbstractChannel.java:259) ~[netty-transport-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.bootstrap.AbstractBootstrap$2.run(AbstractBootstrap.java:366) ~[netty-transport-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163) ~[netty-common-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:405) ~[netty-common-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500) ~[netty-transport-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:906) ~[netty-common-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.36.Final.jar:4.1.36.Final]
	... 1 more
18:42:56.121 [service-container-main] [0.0.0.0:27704-zb-actors-1] ERROR io.zeebe.servicecontainer - Failed to build service
java.lang.RuntimeException: Gateway was not able to start
	at io.zeebe.broker.clustering.base.EmbeddedGatewayService.startGateway(EmbeddedGatewayService.java:64) ~[classes/:?]
	at io.zeebe.util.sched.RunnableAdapter.lambda$wrapRunnable$0(RunnableAdapter.java:48) ~[zeebe-util-0.18.0-SNAPSHOT.jar:0.18.0-SNAPSHOT]
	at io.zeebe.util.sched.RunnableAdapter.run(RunnableAdapter.java:35) ~[zeebe-util-0.18.0-SNAPSHOT.jar:0.18.0-SNAPSHOT]
	at io.zeebe.util.sched.BlockingPollSubscription.run(BlockingPollSubscription.java:55) ~[zeebe-util-0.18.0-SNAPSHOT.jar:0.18.0-SNAPSHOT]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_212]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_212]
	at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_212]
Caused by: java.io.IOException: Failed to bind
	at io.grpc.netty.NettyServer.start(NettyServer.java:251) ~[grpc-netty-1.20.0.jar:1.20.0]
	at io.grpc.internal.ServerImpl.start(ServerImpl.java:177) ~[grpc-core-1.20.0.jar:1.20.0]
	at io.grpc.internal.ServerImpl.start(ServerImpl.java:85) ~[grpc-core-1.20.0.jar:1.20.0]
	at io.zeebe.gateway.Gateway.start(Gateway.java:89) ~[zeebe-gateway-0.18.0-SNAPSHOT.jar:0.18.0-SNAPSHOT]
	at io.zeebe.broker.clustering.base.EmbeddedGatewayService.startGateway(EmbeddedGatewayService.java:62) ~[classes/:?]
	... 6 more
Caused by: java.net.BindException: Address already in use
	at sun.nio.ch.Net.bind0(Native Method) ~[?:1.8.0_212]
	at sun.nio.ch.Net.bind(Net.java:433) ~[?:1.8.0_212]
	at sun.nio.ch.Net.bind(Net.java:425) ~[?:1.8.0_212]
	at sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:223) ~[?:1.8.0_212]
	at io.netty.channel.socket.nio.NioServerSocketChannel.doBind(NioServerSocketChannel.java:132) ~[netty-transport-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.bind(AbstractChannel.java:563) ~[netty-transport-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.bind(DefaultChannelPipeline.java:1332) ~[netty-transport-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeBind(AbstractChannelHandlerContext.java:503) ~[netty-transport-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.channel.AbstractChannelHandlerContext.bind(AbstractChannelHandlerContext.java:488) ~[netty-transport-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.channel.DefaultChannelPipeline.bind(DefaultChannelPipeline.java:984) ~[netty-transport-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.channel.AbstractChannel.bind(AbstractChannel.java:259) ~[netty-transport-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.bootstrap.AbstractBootstrap$2.run(AbstractBootstrap.java:366) ~[netty-transport-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163) ~[netty-common-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:405) ~[netty-common-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500) ~[netty-transport-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:906) ~[netty-common-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.36.Final.jar:4.1.36.Final]
	... 1 more
```

--- a/transport/src/main/java/io/zeebe/transport/impl/util/SocketUtil.java
+++ b/transport/src/main/java/io/zeebe/transport/impl/util/SocketUtil.java
@@ -40,7 +40,18 @@ public class SocketUtil {
     final int testForkNumber = getTestForkNumber();
     final int testMavenId = getTestMavenId();
 
-    final int min = BASE_PORT + testForkNumber * RANGE_SIZE + (testMavenId * RANGE_SIZE * 10);
+    LOG.info(
+        "Starting socket assignment with testForkNumber {} and testMavenId {}",
+        testForkNumber,
+        testMavenId);
+
+    // ensure limits to stay in available port range
+    assert testForkNumber < 39 : "System property test fork number has to be smaller then 39";
+    assert testMavenId < 10 : "System property test maven id has to be smaller then 10";
+
+    final int testOffset = testForkNumber * 10 + testMavenId;
+
+    final int min = BASE_PORT + testOffset * RANGE_SIZE;
     final int max = min + RANGE_SIZE;
     TEST_FORK_NUMBER = testForkNumber;
     PORT_RANGE = new PortRange(DEFAULT_HOST, min, max);
@@ -106,6 +117,7 @@ public class SocketUtil {
     int currentOffset;
 
     PortRange(String host, int min, int max) {
+      assert max <= 65535 : "Port range exceeds maximal available port 65535, got max port " + max;
       this.host = host;
       this.basePort = min;
       this.maxOffset = max - min;

